### DISCUSSION TITLE
Problem with spawning actors from persisted state

### DISCUSSION THREAD

**Discussion by tsxoxo** [statelyai/xstate#4226] [Q&A] [answered]
This seems to be a well-known issue. Hope this can still be of value. Are there any solutions to this at the moment?

---

https://codesandbox.io/p/sandbox/need-8598mq?file=%2Fsrc%2Ftodos.machine.ts%3A5%2C48

---

I'm working on adding persistent storage to the official [Todo-MVC app for Xstate + Vue](https://xstate.js.org/docs/examples/todomvc.html#todomvc-with-vue). 

When I try to rehydrate from `localStorage`, I get 'dead actors' that don't send events back to the parent.

See screenshot below. 8 is a "retrieved" actors from storage, 9 is a fresh one that was added on the client.

![image](https://github.com/statelyai/xstate/assets/59713582/55536cf3-2329-4f2b-9d7d-cd24ec8f46c4)

I put "retrieved" in quotes, because reading the code in `todosMachine.state.loading.entry` it looks like it should generate a fresh actor, but it doesn't.

I'd also love to know some ways to debug something like this -- is there a way to show all spawned actors that are 'running'?

# Also tried
Doing it with `useMachine(todos, {state: persistedState}` like [the `@xstate/vue` docs suggest](https://xstate.js.org/docs/packages/xstate-vue/#persisted-and-rehydrated-state).

`const { state, send } = useMachine(todosMachine, { context: contextFromLocalStorage });`

result: same behaviour

# Possibly related
#4171 #4147 #4148 #1476 #3962

**Comment by davidkpiano** [comment, marked as answer]
Persisting invoked actors is solved in XState v5 and will be solved for spawned actors in XState v5. There are no plans to make this work for v4 due to complexity of the solution.

https://github.com/statelyai/xstate/pull/3921

**Comment by tsxoxo** [comment]
Ok, so I'd wanna migrate my project to v5. Two questions:

* would this break the `@xstate/vue` package?
* would y'all be interested in me submitting a PR?

**Reply by davidkpiano** [reply]
The `@xstate/vue@beta` package should be (+ will be #4288) up to date with `xstate@beta`, if that's what you're asking.
